holoviews/plotting/bokeh/callbacks.py

```diff
@@ -215,13 +279,18 @@ def _filter_msg(self, msg, ids):
 
     def on_msg(self, msg):
         for stream in self.streams:
-            ids = self.handle_ids[stream]
+            metadata = self.handle_ids[stream]
+            ids = list(metadata.values())
             filtered_msg = self._filter_msg(msg, ids)
             processed_msg = self._process_msg(filtered_msg)
             if not processed_msg:
                 continue
             stream.update(trigger=False, **processed_msg)
+            stream._metadata = {h: {'id': hid, 'events': self.events}
+                                for h, hid in metadata.items()}
         Stream.trigger(self.streams)
+        for stream in self.streams:
+            stream._metadata = None
```

Comment:
I would consider just declaring ``_metadata`` to be an empty dict by default. I'm assuming this loop is to reset...